distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/cli/SyncPluginsAction.java

```diff
@@ -207,9 +208,10 @@ private List<InstallablePlugin> getPluginsToUpgrade(
         Optional<PluginsConfig> cachedPluginsConfig,
         List<PluginDescriptor> existingPlugins
     ) {
-        final Map<String, String> cachedPluginIdToLocation = cachedPluginsConfig.map(
-            config -> config.getPlugins().stream().collect(Collectors.toMap(InstallablePlugin::getId, InstallablePlugin::getLocation))
-        ).orElse(Map.of());
+        final Map<String, String> cachedPluginIdToLocation = new HashMap<>();
+        cachedPluginsConfig.ifPresent(
+            config -> config.getPlugins().forEach(p -> cachedPluginIdToLocation.put(p.getId(), p.getLocation()))
```

Comment:
Could this be `cachedPluginIdToLocation.putAll(config.getPlugins())`?

Reply:
config.getPlugins() returns a List, while putAll accepts a Map.
I can use the basic `collect` like
```
config.getPlugins().stream().collect(HashMap::new, (m,v)->m.put(v.getId(), v.getLocation()), HashMap::putAll) 
```
And keep it closer to the original code, but using `forEach` and `put` seemed simpler (and probably more efficient).
Wdyt?

Reply:
What you have is fine then. I was hoping to avoid such nested lambdas.